### PR TITLE
congestion: runtime test for outgoing buffer

### PR DESCRIPTION
**Pull request by jakmeier**
Test on Runtime::apply level that buffering and forwarding works.

This is complementary to unit tests in congestion_info.rs,
balance_checker.rs,  receipt_column_helper.rs, and of course
integrations tests.

### CODE REVIEW

**Comment by jakmeier** [comment]
In draft until #11300 is merged, which this depends on for tests to pass.

**Comment by jakmeier** [comment]
This is ready for review, in theory. But I would rather complete and merge #11307 first and adjust this PR to use the config values. 
It probably makes sense to wait with reviewing it until then.

File: runtime/runtime/src/lib.rs [hunk 1]
```diff
@@ -1820,7 +1822,7 @@ impl Runtime {
             proof,
             delayed_receipts_count: delayed_receipts.len(),
             metrics: Some(metrics),
-            congestion_info: congestion_info.map(|info| info.congestion_info),
+            congestion_info: congestion_control.map(|c| *c.congestion_info()),
```
Comment (jakmeier) [inline runtime/runtime/src/lib.rs:1825]: This fixes a bug introduced in #11307.
We must return the modified congestion info, not the input congestion info.

The tests added by in this PR catches it, while existing tests didn't.

**Comment by jakmeier** [comment]
@wacban this is now ready for review, using the new runtime parameters and updated to use the new interface. Plus a small fix that slipped in and previous tests didn't cover.

**Review by wacban — APPROVED** [review]
LGTM

**Pull request merged**